builtin/logical/database/path_config_connection.go

```diff
@@ -214,19 +232,46 @@ func (b *databaseBackend) connectionDeleteHandler() framework.OperationFunc {
 // both builtin and plugin database types.
 func (b *databaseBackend) connectionWriteHandler() framework.OperationFunc {
 	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
-		pluginName := data.Get("plugin_name").(string)
-		if pluginName == "" {
-			return logical.ErrorResponse(respErrEmptyPluginName), nil
-		}
+		verifyConnection := data.Get("verify_connection").(bool)
 
 		name := data.Get("name").(string)
 		if name == "" {
 			return logical.ErrorResponse(respErrEmptyName), nil
 		}
 
-		verifyConnection := data.Get("verify_connection").(bool)
-		allowedRoles := data.Get("allowed_roles").([]string)
-		rootRotationStatements := data.Get("root_rotation_statements").([]string)
+		// Baseline
+		config := &DatabaseConfig{}
+
+		entry, err := req.Storage.Get(ctx, fmt.Sprintf("config/%s", name))
+		if err != nil {
+			return nil, errors.New("failed to read connection configuration")
+		}
+		if entry != nil {
+			if err := entry.DecodeJSON(config); err != nil {
+				return nil, err
+			}
+		}
+
+		if pluginNameRaw, ok := data.GetOk("plugin_name"); ok {
```

Comment:
I _think_ you can change the condition to: `ok || req.Operation == logical.CreateOperation` and remove the `else if`?  Same for the similar blocks below this.